From a555497d6bf9cb939e75e406f1538740e4f4ff06 Mon Sep 17 00:00:00 2001
From: Matheus <sirherobrine23@users.noreply.github.com>
Date: Fri, 24 Jul 2026 01:29:46 +0000
Subject: [PATCH] ARM: dts: airoha: split EN7523 SCU nodes

Represent the chip SCU and NP-SCU clock/reset controller as separate
syscon nodes. The EN7523 clock driver already supports this split layout
through the airoha,chip-scu phandle and named np-scu and mcucfg resources.

Signed-off-by: Matheus <sirherobrine23@users.noreply.github.com>
Signed-off-by: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
---
 arch/arm/boot/dts/airoha/en7523.dtsi | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/airoha/en7523.dtsi b/arch/arm/boot/dts/airoha/en7523.dtsi
index 115cc4409..6d142d3a4 100644
--- a/arch/arm/boot/dts/airoha/en7523.dtsi
+++ b/arch/arm/boot/dts/airoha/en7523.dtsi
@@ -87,10 +87,17 @@ L2_0: l2-cache0 {
 		};
 	};
 
-	scu: system-controller@1fa20000 {
-		compatible = "airoha,en7523-scu";
-		reg = <0x1fa20000 0x400>,
-		      <0x1fb00000 0x1000>;
+	chip_scu: syscon@1fa20000 {
+		compatible = "airoha,chip-scu", "syscon";
+		reg = <0x1fa20000 0x400>;
+	};
+
+	scuclk: system-controller@1fb00000 {
+		compatible = "airoha,en7523-scu", "airoha,scuclk", "syscon";
+		reg = <0x1fb00000 0x1000>,
+		      <0x1efbe000 0x800>;
+		reg-names = "np-scu", "mcucfg";
+		airoha,chip-scu = <&chip_scu>;
 		#clock-cells = <1>;
 		#reset-cells = <1>;
 	};
-- 
2.55.0

